File: ScaleformUI_Csharp/Panels/UIMenuStatisticsPanel.cs
using System.Collections.Generic;
using System.Drawing;
using System.Threading.Tasks;
using CitizenFX.Core.UI;

namespace ScaleformUI
{
	public class UIMenuStatisticsPanel : UIMenuPanel
	{
		internal readonly List<StatisticsForPanel> Items;

		public UIMenuStatisticsPanel()
		{
			Items = new List<StatisticsForPanel>();
		}

		public void AddStatistics (string Name, float val)
		{
			float _value = val;
			if (_value > 100)
				_value = 100;
			if (_value < 0)
				_value = 0;
			StatisticsForPanel item = new(Name, _value);
			Items.Add(item);
			var it = ParentItem.Parent.MenuItems.IndexOf(ParentItem);
			var van = ParentItem.Panels.IndexOf(this);
			ScaleformUI._ui.CallFunction("ADD_STATISTIC_TO_PANEL", it, van, Name, _value);
		}

		public float GetPercentage(int ItemId)
		{
			return Items[ItemId].Value;
		}

		public void UpdateStatistic(int ItemId, float number)
		{
			Items[ItemId].Value = number;
			if (Items[ItemId].Value > 100)
				Items[ItemId].Value = 100;
			if (Items[ItemId].Value < 0)
				Items[ItemId].Value = 0;
			var it = ParentItem.Parent.MenuItems.IndexOf(ParentItem);
			var van = ParentItem.Panels.IndexOf(this);
			ScaleformUI._ui.CallFunction("SET_PANEL_STATS_ITEM_VALUE", it, van, ItemId, Items[ItemId].Value);
		}
	}

	public class StatisticsForPanel
	{
		public string Text { get; set; }
		public float Value { get; set; }

		public StatisticsForPanel(string label, float value)
        {
			Text = label;
			Value = value;
			if (Value > 100)
				Value = 100;
			if(Value < 0)
				Value = 0;
        }
	}
}

File: ScaleformUI_V8/src/menus/UIMenu/panels/uimenustatisticspanel.ts
import { ScaleformUI } from "scaleforms/scaleformui/main";
import { UIMenuPanel } from "./uimenupanel";

export class UIMenuStatisticsPanel extends UIMenuPanel {
    Items: StatisticsForPanel[];

    constructor() {
        super();
        this.Items = [];
    }

    public AddStatistic(name: string, val: number) {
        let _value = val;
        if (_value > 100)
            _value = 100;
        if (_value < 0)
            _value = 0;
        let item = new StatisticsForPanel(name, _value);
        this.Items.push(item)
        if (this.ParentItem != null && this.ParentItem.Parent != null && this.ParentItem.Parent.Visible) {
            let it = this.ParentItem.Parent.Pagination.GetScaleformIndex(this.ParentItem.Parent.Items.indexOf(this.ParentItem));
            let van = this.ParentItem.Panels.indexOf(this);
            ScaleformUI.Scaleforms._ui?.callFunction("ADD_STATISTIC_TO_PANEL", it, van, name, _value);
        }
    }

    public GetPercentage(ItemId: number): number {
        return this.Items[ItemId].Value;
    }

    public UpdateStatistic(itemId: number, value: number) {
        this.Items[itemId].Value = value;
        if (this.Items[itemId].Value > 100)
            this.Items[itemId].Value = 100;
        if (this.Items[itemId].Value < 0)
            this.Items[itemId].Value = 0;
        let it = this.ParentItem.Parent.Pagination.GetScaleformIndex(this.ParentItem.Parent.Items.indexOf(this.ParentItem));
        let van = this.ParentItem.Panels.indexOf(this);
        ScaleformUI.Scaleforms._ui?.callFunction("SET_PANEL_STATS_ITEM_VALUE", it, van, itemId, this.Items[itemId].Value);
    }

}

export class StatisticsForPanel {
    public Text: string;
    public Value: number;

    constructor(label: string, value: number) {
        this.Text = label;
        this.Value = value;
        if (this.Value > 100)
            this.Value = 100;
        if (this.Value < 0)
            this.Value = 0;
    }
}